### PR TITLE
Added ID restoration terminal

### LINKED ISSUE #3748: There needs to be an easier way to fix gurgled IDs
#### Brief description of the issue
If ID gets gurgled right now the only way to fix it is by having CD or HoP around. Which does NOT happen often during deadshift hours. Or sometimes during normal hours. Or they got eaten. Or they're in a scene. Limitedness and command position of those jobs make it even worse. Also, they rarely actually reply to calls on bulletin just to 'fix ID'. In the end you get stuck with useless ID, pretty much demoted for the rest of the round even if your digestion was willing, with itemweak and in general meant to be completely harmless.

#### What you expected to happen
To be able to access your department after getting resleeved post-digestion

#### What actually happened
Got fucked over and ended up sitting afk unable to do shit for the entire rest of the shift

#### Steps to reproduce
- Get gurgled
- Get your gurgled ID
- Be sad that you're now visitor because no HoP/CD or they don't answer.

#### Code Revision
- c973f8672a981994941b90fc583c88c1ddab5fac 
 
#### Anything else you may wish to add:
While I understand sentiment behind pred not being able to eat people just to get their access, this fucks over people big time. Imagine only CD or HoP getting gurgled? Or someone on shift where neither are present and its a night deadshift, so one being present is blissful dream? Resleeved prey getting punished over something meant to hinder digesty preds is not good, IMO. Especially since if anything bad happened that shift I lost my access, I wouldn't be able to go fix it, because RIP my engineering access as the only engineer.

### PR DESCRIPTION
**Pull request by Heroman3003**
What it says on the tin. Fixes contaminated ID's access codes. Does not work if ID has been modified in ID console after digestion. Only works if person trying to restore access is identifiably ID's owner (or looks exactly like them).

Only one exists, its mapped in near HoP's office. Looks like green version of guestpass console.

EDIT1: nearly forgot, Fixes #3748

EDIT2: Now with unique icons and icon for successful/failing restoration. Icons provided by everyone's favorite VerySoft.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
I'm okay with this.

File: code/game/machinery/computer/id_restorer_vr.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+				return
+			var/mob/living/carbon/human/H = user
+			if(!(istype(H)))
+				to_chat(user, "<span class='warning'>Inappropriate species interfacing. Access denied.</span>")
+				return
+			else if((H.wear_mask && (H.wear_mask.flags_inv & HIDEFACE)) || (H.head && (H.head.flags_inv & HIDEFACE)))	//Face hiding bad
+				to_chat(user, "<span class='warning'>Face scan obstructed. Access denied.</span>")
+				return
+			else if(H.get_face_name() == "Unknown" || !(H.real_name == inserted.registered_name))
+				to_chat(user, "<span class='warning'>Face scan failed Access denied.</span>")
```
Comment (SpadesNeil) [inline code/game/machinery/computer/id_restorer_vr.dm]: Grammar problems, and also I'd like it to say `Facial recognition scan failed. Access denied.`

Comment (Heroman3003) [reply code/game/machinery/computer/id_restorer_vr.dm]: Feel free to correct any error messages you'd prefer differently worded. As long as they convey why its denying, it doesn't really matter.

**Comment by Wickedtemp** [comment]
Can you elaborate on the bit about it not working if modified? 

Is that just a thing that blocks the restoration if the ID had been previously changed in the HoP's console?

**Comment by Screemonster** [comment]
> Can you elaborate on the bit about it not working if modified?
> 
> Is that just a thing that blocks the restoration if the ID had been previously changed in the HoP's console?

If the card is damaged and subsequently modified in an ID console, the ID console overwrites the previous "lost" access.

**Comment by Very-Soft** [comment]
idle
success
fail

**Comment by Heroman3003** [comment]
Aaand icons implemented!

**Pull request merged**